### ISSUE TITLE
yolo26n ncnn export

### ISSUE THREAD

**Issue by wanghongfei833** [ultralytics/ultralytics#23365]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

When I try to export YOLOv26 to ncnn format using the latest version of ultralytics, I encounter the following warning and cannot successfully achieve end-to-end export.
<img width="1679" height="179" alt="Image" src="https://github.com/user-attachments/assets/cab72eef-3ba5-4dc0-8210-6c4518ae682a" />
Switching to ultralytics==8.4.0 allows me to export the model smoothly, but the exported model fails to perform proper inference.

<img width="2078" height="1957" alt="Image" src="https://github.com/user-attachments/assets/b83cde0b-cebb-429e-bd40-974452be4567" />

<img width="2010" height="1562" alt="Image" src="https://github.com/user-attachments/assets/8d73b58b-1cfd-48a6-9498-7abb28e51921" />

**Comment by UltralyticsAssistant** [comment]
👋 Hello @wanghongfei833, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

Thanks for sharing the screenshots—NCNN export can be sensitive to versions and export settings. This is an automated response 🤖, and an Ultralytics engineer will also assist soon.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it. In your case, an MRE that includes the exact export command/code you used for YOLO26n, plus the minimal steps you used to validate inference on the exported NCNN model, would help a lot.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by onuralpszr** [comment]
@wanghongfei833  could you please try with ultralytics version 8.4.7 (latest) with export and inference. 

My test results

```
 > yolo export model="yolo26n.pt" format="ncnn"
Ultralytics 8.4.7 🚀 Python-3.12.12 torch-2.9.1+cu128 CPU (13th Gen Intel Core(TM) i7-13850HX)
WARNING ⚠ NCNN export does not support end2end models, disabling end2end branch.
YOLO26n summary (fused): 136 layers, 2,490,772 parameters, 0 gradients, 5.4 GFLOPs

PyTorch: starting from 'yolo26n.pt' with input shape (1, 3, 640, 640) BCHW and output shape(s) (1, 84, 8400) (5.3 MB)
requirements: Ultralytics requirement ['pnnx'] not found, attempting AutoUpdate...
Resolved 27 packages in 1.07s
Downloading pnnx (26.4MiB)
 Downloaded pnnx
Prepared 1 package in 3.60s
warning: Failed to hardlink files; falling back to full copy. This may lead to degraded performance.
         If the cache and target directories are on different filesystems, hardlinking may not be supported.
         If this is intentional, set `export UV_LINK_MODE=copy` or use `--link-mode=copy` to suppress this warning.
Installed 1 package in 27ms
 + pnnx==20260112

requirements: AutoUpdate success ✅ 4.8s
WARNING ⚠ requirements: Restart runtime or rerun command for updates to take effect


NCNN: starting export with NCNN 1.0.20260114 and PNNX 20260112...
pnnxparam = yolo26n_ncnn_model/model.pnnx.param
pnnxbin = yolo26n_ncnn_model/model.pnnx.bin
pnnxpy = yolo26n_ncnn_model/model_pnnx.py
pnnxonnx = yolo26n_ncnn_model/model.pnnx.onnx
ncnnparam = yolo26n_ncnn_model/model.ncnn.param
ncnnbin = yolo26n_ncnn_model/model.ncnn.bin
ncnnpy = yolo26n_ncnn_model/model_ncnn.py
fp16 = 0
optlevel = 2
device = cpu
inputshape = [1,3,640,640]f32
inputshape2 = 
customop = 
moduleop = 
get inputshape from traced inputs
inputshape = [1,3,640,640]f32
############# pass_level0
inline module = torch.nn.modules.linear.Identity
inline module = ultralytics.nn.modules.block.Attention
inline module = ultralytics.nn.modules.block.Bottleneck
inline module = ultralytics.nn.modules.block.C2PSA
inline module = ultralytics.nn.modules.block.C3k
inline module = ultralytics.nn.modules.block.C3k2
inline module = ultralytics.nn.modules.block.PSABlock
inline module = ultralytics.nn.modules.block.SPPF
inline module = ultralytics.nn.modules.conv.Concat
inline module = ultralytics.nn.modules.conv.Conv
inline module = ultralytics.nn.modules.conv.DWConv
inline module = ultralytics.nn.modules.head.Detect
inline module = torch.nn.modules.linear.Identity
inline module = ultralytics.nn.modules.block.Attention
inline module = ultralytics.nn.modules.block.Bottleneck
inline module = ultralytics.nn.modules.block.C2PSA
inline module = ultralytics.nn.modules.block.C3k
inline module = ultralytics.nn.modules.block.C3k2
inline module = ultralytics.nn.modules.block.PSABlock
inline module = ultralytics.nn.modules.block.SPPF
inline module = ultralytics.nn.modules.conv.Concat
inline module = ultralytics.nn.modules.conv.Conv
inline module = ultralytics.nn.modules.conv.DWConv
inline module = ultralytics.nn.modules.head.Detect

----------------

############# pass_level1
############# pass_level2
############# pass_level3
############# pass_level4
############# pass_level5
############# pass_ncnn
NCNN: export success ✅ 6.6s, saved as 'yolo26n_ncnn_model' (9.4 MB)

Export complete (6.7s)
Results saved to ultralytics/
Predict:         yolo predict task=detect model=yolo26n_ncnn_model imgsz=640 
Validate:        yolo val task=detect model=yolo26n_ncnn_model imgsz=640 data=/home/lq/codes/ultralytics/ultralytics/cfg/datasets/coco.yaml  
Visualize:       https://netron.app
💡 Learn more at https://docs.ultralytics.com/modes/export
```

CLI Inference Test

```bash
❯ yolo predict task=detect model=yolo26n_ncnn_model imgsz=640 
WARNING ⚠ 'source' argument is missing. Using default 'source=.venv/lib/python3.12/site-packages/ultralytics/assets'.
Ultralytics 8.4.7 🚀 Python-3.12.12 torch-2.9.1+cu128 CUDA:0 (NVIDIA RTX 4000 Ada Generation Laptop GPU, 11874MiB)
Loading yolo26n_ncnn_model for NCNN inference...

image 1/2 .venv/lib/python3.12/site-packages/ultralytics/assets/bus.jpg: 640x640 4 persons, 1 bus, 46.0ms
image 2/2 .venv/lib/python3.12/site-packages/ultralytics/assets/zidane.jpg: 640x640 2 persons, 1 tie, 37.5ms
Speed: 2.9ms preprocess, 41.8ms inference, 0.7ms postprocess per image at shape (1, 3, 640, 640)
Results saved to ultralytics/runs/detect/predict40
💡 Learn more at https://docs.ultralytics.com/modes/predict
```

Python API Test

```python
from ultralytics import YOLO
model = YOLO(model="yolo26n_ncnn_model", task="detect")
model.predict(source="bus.jpg")
```

```bash
Loading yolo26n_ncnn_model for NCNN inference...

image 1/1 /home/osezer/ultralytics-working-folder/ultralytics-develop/bus.jpg: 640x640 4 persons, 1 bus, 54.1ms
Speed: 3.3ms preprocess, 54.1ms inference, 0.9ms postprocess per image at shape (1, 3, 640, 640)
```

**Comment by wanghongfei833** [comment]
Yes, after updating to the latest version, I can successfully export the model and perform inference. However, the warning message:

"WARNING ⚠ NCNN export does not support end2end models, disabling end2end branch."​

still persists. I've checked the output shape of the exported model, and it remains 1x84x8000. Is it possible to achieve an output shape of 1x300x6 instead?

**Comment by Y-T-G** [comment]
@wanghongfei833 

NCNN doesn't support the end2end head, so you can't get an inference result like (1,300,6).

But it's possible to avoid NMS by performing the postprocessing for the end2end model manually.
